internal/proxy/proxy.go

```diff
@@ -133,32 +143,76 @@ func NewClient(ctx context.Context, d cloudsql.Dialer, cmd *cobra.Command, conf
 		}(inst.Name)
 	}
 	pc := newPortConfig(conf.Port)
+
 	for _, inst := range conf.Instances {
-		m := &socketMount{inst: inst.Name}
-		a := conf.Addr
-		if inst.Addr != "" {
-			a = inst.Addr
-		}
 		version, err := d.EngineVersion(ctx, inst.Name)
 		if err != nil {
 			return nil, err
 		}
-		var np int
-		switch {
-		case inst.Port != 0:
-			np = inst.Port
-		case conf.Port != 0:
-			np = pc.nextPort()
-		default:
-			np = pc.nextDBPort(version)
+
+		var (
+			// network is one of "tcp" or "unix"
+			network string
+			// address is either a TCP host port, or a Unix socket
+			address string
+		)
+		if (conf.UnixSocket != "" || inst.UnixSocket != "") &&
+			(inst.Addr == "" && inst.Port == 0) {
+			network = "unix"
+
+			dir := conf.UnixSocket
+			if dir == "" {
+				dir = inst.UnixSocket
+			}
+			// Attempt to make the directory if it does not already exist.
+			if _, err := os.Stat(dir); err != nil {
+				if err = os.Mkdir(dir, 0750); err != nil {
```

Comment:
Why these file permissions? It looks like do this inconsistently in the proxy, and use both `0755` and `0777` in different places.

Reply:
We should probably should make these consistent elsewhere. I think we'd want to allow the user read-write-execute permissions (7), group might deserve the same or even just read permissions (5), and other seems like they shouldn't have any permissions whatsoever (0). So, for any Unix socket or Fuse file, I'd expect either 750, or even 700, although I'm less certain what the right group permission should be.

Reply:
Did some research on this. https://www.postgresql.org/docs/current/runtime-config-connection.html#:~:text=The%20default%20permissions%20are%200777,revoking%20read%20or%20execute%20permissions.)

In short, the default is 0777 with 0770 being an acceptable alternative. Since there's a chance some users rely on file permissions to control access to the database, I'm going to stick with user and group access, but no access for other, i.e., 0770.